extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/SmallRyeReactiveMessagingProcessor.java

```diff
@@ -82,15 +84,24 @@ void validateBeanDeployment(
             }
         }
 
-        for (InjectionPointInfo injectionPoint : validationPhase.getContext().get(BuildExtension.Key.INJECTION_POINTS)) {
+        for (InjectionPointInfo injectionPoint : validationPhase.getContext()
+                .get(BuildExtension.Key.INJECTION_POINTS)) {
             if (injectionPoint.getRequiredType().name().equals(NAME_EMITTER)) {
                 AnnotationInstance stream = injectionPoint.getRequiredQualifier(NAME_STREAM);
                 if (stream != null) {
                     // Stream.value() is mandatory
                     String name = stream.value().asString();
+                    AnnotationInstance annotation = injectionPoint.getTarget().asField()
+                            .annotation(ON_OVERFLOW);
+
                     LOGGER.debugf("Emitter injection point '%s' detected, stream name: '%s'",
                             injectionPoint.getTargetInfo(), name);
-                    emitters.produce(new EmitterBuildItem(name));
+                    EmitterBuildItem item = new EmitterBuildItem(name);
+                    if (annotation != null) {
+                        item.setOverflow(annotation.value().toString());
```

Comment:
Ok now I see why those setters. Well, I'd personally prefer a static factory method for this particular use case, ie. something like `EmitterBuildItem.from(name, annotation)` and final fields.

Reply:
+1